src/sentry/api/endpoints/project_group_index.py

```diff
@@ -703,18 +710,18 @@ def put(self, request, project):
                         activity.send_notification()
 
         if 'assignedTo' in result:
-            if result['assignedTo']:
+            assigned_actor = result['assignedTo']
+            if assigned_actor:
                 for group in group_list:
-                    GroupAssignee.objects.assign(
-                        group, result['assignedTo'], acting_user)
+                    resolved_actor = assigned_actor.resolve()
+                    GroupAssignee.objects.assign(group, resolved_actor, acting_user)
 
-                    if 'isSubscribed' not in result or result['assignedTo'] != request.user:
-                        GroupSubscription.objects.subscribe(
-                            group=group,
-                            user=result['assignedTo'],
-                            reason=GroupSubscriptionReason.assigned,
-                        )
-                result['assignedTo'] = serialize(result['assignedTo'])
+                    GroupSubscription.objects.subscribe_actor(
+                        group=group,
+                        actor=resolved_actor,
+                        reason=GroupSubscriptionReason.assigned,
+                    )
+                result['assignedTo'] = serialize(assigned_actor.get_actor_id())
```

Comment:
Why are we just serializing a string? Don't we want to serialize the `assigned_actor` object?